src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -136,7 +146,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     } catch (InterruptedException | RejectedExecutionException e) {
       allCards = Collections.emptyList();
     }
-
+    if (!preferredSite.isEmpty()) {
+      List<Card> sortedCards = allCards;
+      for (Card card : sortedCards) {
+        if (card.getSource().toLowerCase().equals(preferredSite)) {
+          int index = allCards.indexOf(card);
+          sortedCards.remove(index);
```

Comment:
add `if (index != 0)` here, if the index already equals to 0, no further action is needed